client/src/components/DataFiles/DataFilesModals/DataFilesMoveModal.jsx

```diff
@@ -12,74 +18,53 @@ const DataFilesMoveModal = React.memo(() => {
   const history = useHistory();
   const location = useLocation();
 
+  const { params } = useFileListing('FilesListing');
+  const {
+    params: modalParams,
+    data: files,
+    fetchListing,
+  } = useFileListing('modal');
+
+  const { move, status, setStatus } = useMove();
+
   const dispatch = useDispatch();
-  const params = useSelector(
-    (state) => state.files.params.FilesListing,
-    shallowEqual
-  );
-  const modalParams = useSelector(
-    (state) => state.files.params.modal,
-    shallowEqual
-  );
+  const { data: systems } = useSystems();
 
   const reloadPage = () => {
     history.push(location.pathname);
-    dispatch({
-      type: 'FETCH_FILES_MODAL',
-      payload: { ...modalParams, section: 'modal' },
-    });
+    fetchListing(modalParams);
   };
 
-  const files = useSelector((state) => state.files.listing.modal, shallowEqual);
-  const isOpen = useSelector((state) => state.files.modals.move);
-  const selectedFiles = useSelector((state) =>
-    state.files.selected.FilesListing.map((i) => ({
-      ...state.files.listing.FilesListing[i],
-      id: uuidv4(),
-    }))
-  );
+  const { getStatus, toggle: toggleModal } = useModal();
+
+  const isOpen = getStatus('move');
+  const { selectedFiles } = useSelectedFiles();
   const selected = useMemo(() => selectedFiles, [isOpen]);
-  const status = useSelector(
-    (state) => state.files.operationStatus.move,
-    shallowEqual
-  );
   const [disabled, setDisabled] = useState(false);
 
-  const toggle = () =>
-    dispatch({
-      type: 'DATA_FILES_TOGGLE_MODAL',
-      payload: { operation: 'move', props: {} },
-    });
+  const toggle = () => toggleModal({ operation: 'move', props: {} });
 
   const onOpened = () => {
-    dispatch({
-      type: 'FETCH_FILES_MODAL',
-      payload: { ...params, section: 'modal' },
+    fetchListing({
+      api: 'tapis',
+      scheme: 'private',
+      system: systems[0].system,
     });
   };
 
   const onClosed = () => {
     dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
-    dispatch({
-      type: 'DATA_FILES_SET_OPERATION_STATUS',
-      payload: { operation: 'move', status: {} },
-    });
+    setStatus({});
     setDisabled(false);
   };
 
   const moveCallback = useCallback(
```

Comment:
Can you remind me why the move and copy callbacks use `useCallback`?

Reply:
Most likely premature optimization when we were first getting the hang of react/hooks. There's supposed to be a performance benefit from memoizing function definitions, but it's probably negligible in practice.